viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -19,6 +19,44 @@
 #include "cartographer/mapping/map_builder.h"
 #include "glog/logging.h"
 
+namespace {
+// The default value when coloring the image if no data is present. A
+// pixel representing no data will have all 3 channels of its color channels
+// (RGB) as the default value([102,102,102])
+static const unsigned char defaultNoDataRGBValue = 102;
+// Number of bytes in a pixel
+static const int bytesPerPixel = 4;
+struct ColorARGB {
+    unsigned char A;
+    unsigned char R;
+    unsigned char G;
+    unsigned char B;
+};
+
+// Check if pixel is the default color signaling no data is present
+bool CheckIfEmptyPixel(ColorARGB pixel_color) {
+    return (pixel_color.R == defaultNoDataRGBValue) &&
+           (pixel_color.G == defaultNoDataRGBValue) &&
+           (pixel_color.B == defaultNoDataRGBValue);
+}
+
+// Conver the scale of a specified pixel color channel from the given
+// 102-255 range to 100-0. This is an initial solution for extracting
+// probability information from cartographer
+int CalculateProbabilityFromColorChannels(ColorARGB pixel_color) {
+    unsigned char max_val = CHAR_MAX;
+    unsigned char min_val = defaultNoDataRGBValue;
+    unsigned char max_prob = 100;
+    unsigned char min_prob = 0;
+
+    // Probability is currently determined solely by the R channel
+    unsigned char color_channel_val = pixel_color.R;
+    unsigned char prob = (max_val - color_channel_val) * (max_prob - min_prob) /
```

Comment:
Are we certain `color_channel_val` can't be lower than 102?

If we are then I'm good with leaving this as is. 

If we are not certain but want to represent any color channel value equal to, or below 102 as probability 0, I recommend doing the following change to prevent returning probabilities outside of 0 - 100
```suggestion
    unsigned char prob = (max_val - std::max(color_channel_val, min_val)) * (max_prob - min_prob) /
```

Reply:
@JohnN193 question for you John. I recorded the minimum value for pixel and it was 7 not 102. Does this need to be changed? Or is there a reason we would want to threshold it at 102.

Reply:
If you are seeing the probability channel register numbers below 102 consistently then feel free to change.